core/cap-0034.md

```diff
@@ -49,6 +52,80 @@ dependent transactions would then perform operations which the smart contract
+if and only if `T1` executes; the smart contract is depending upon `T2` having
+the wrong sequence number if `T1` is not applied (and therefore does not consume
+a sequence number).  The smart contract sets an expiration time on `T1` (as our
+[documentation](https://www.stellar.org/developers/guides/concepts/transactions.html)
+"highly advises" that clients do).
+
+2. The ledger happens first to have capacity to accept `T1` into a transaction
+set between `T1`'s expiration time and the time at which it will turn out (of
+course the network can not foresee that this will eventually be the case) that
+the next ledger will close.
+
+3. `T1` happens at that point to be in the transaction queue of the validator
+who (though again this is not predictable yet) will turn out to win nomination.
+That node builds a transaction set that contains `T1`.
+
+4. The transaction set containing `T1` wins nomination, but (at least) one of
+the candidate `closeTime`s is greater than `T1`'s expiration time.
+
+5. Transactions are applied using the maximum of all candidate `closeTime`s as
+the new last ledger close time.  `T1` therefore returns `txTOO_LATE` and is
+not applied.  However, as it reached ledger close before being failed, it is
+charged a fee and consumes a sequence number.
+
+6. `T2` gets into an externalized transaction set (possibly, but not
+necessarily, the same one that `T1` got into) before it too expires.  It
+has the correct sequence number, because `T1` consumed a sequence number.  It
+is therefore applied -- even though its preconditions were intended to include
+guarantees that `T1` had made by executing, so the postconditions of `T2`
+could violate any of the smart contract's intended invariants which had
+depended upon `T1` having executed before `T2` could so so.
+
+With the behavior proposed in this CAP, step #3 -- a node building a transaction
+set containing `T1` -- would only occur if `T1` were not expired with respect to
+the `closeTime` in the same proposed `StellarValue` as that transaction set.
+Otherwise, `T1` would not make it into a transaction set, and would therefore
+never consume a sequence number (or be charged a fee), and `T2` would fail
+validation with a bad sequence number, as the smart contract intended in that
+case.  If `T1` did make it into a transaction set, then, under the new behavior
+in this CAP, step #5 would change -- if the transaction set containing `T1` won
+nomination, then the `closeTime` of the new ledger would be the `closeTime` from
+the same `StellarValue` as `T1`, with respect to which `T1` is in this case not
+expired.  Hence, `T1` would be applied during ledger close; it would not return
+`txTOO_LATE`.  `T2` would therefore have the opportunity to execute (assuming it
+was valid in the other respects in addition to its sequence number), and in this
+case that would be as expected, as `T1` had previously executed.
+
+Note that the duration that this race window in the current protocol remains
```

Comment:
This is a very interesting perspective, but definitely better suited for the rationale.

Reply:
As described in a comment above, I've moved that section to (a new sub-section of) the "Design Rationale".